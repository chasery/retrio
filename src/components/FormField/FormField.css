.FormField {
  display: flex;
  flex-direction: column;
}
.FormField + .FormField {
  margin-top: 0.5rem;
}

.FormField__label {
  color: #474747;
}

.FormField__input {
  margin-top: 0.125rem;
  padding: 0.5rem 0.75rem;
  font-size: 1rem;
  font-weight: 500;
  border-radius: 0.125rem;
  border: 0.0625rem solid #ebebeb;
  color: #474747;
  background: #ffffff;
}

.FormField__required {
  padding-left: 0.25rem;
  color: crimson;
}
