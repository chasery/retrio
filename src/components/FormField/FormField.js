import React from "react";
import "./FormField.css";

function FormField(props) {
  const { id, label, type, isRequired, onChange, value } = props;

  return (
    <div className="FormField">
      <label className="FormField__label" htmlFor={id}>
        {label}
        {isRequired ? <span className="FormField__required">*</span> : ""}
      </label>
      <input
        className="FormField__input"
        type={type}
        id={id}
        required={isRequired}
        onChange={onChange}
        value={value}
      />
    </div>
  );
}

export default FormField;
