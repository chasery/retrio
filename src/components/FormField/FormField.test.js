import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import FormField from "./FormField";

describe("FormField component", () => {
  it("renders without error", () => {
    const wrapper = shallow(<FormField />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
