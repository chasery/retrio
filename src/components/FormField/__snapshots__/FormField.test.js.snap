// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField component renders without error 1`] = `
<div
  className="FormField"
>
  <label
    className="FormField__label"
  />
  <input
    className="FormField__input"
  />
</div>
`;
